src/dlgTriggerEditor.cpp

```diff
@@ -3769,111 +4646,91 @@ void dlgTriggerEditor::slot_trigger_selected(QTreeWidgetItem* pItem)
+            pItem->lineEdit_pattern->setPalette(palette);
+            pItem->lineEdit_pattern->setText(patternList[i]);
         }
         for (int i = patternList.size(); i < 50; i++) {
-            mTriggerPatternEdit[i]->lineEdit->clear();
-            if (mTriggerPatternEdit[i]->lineEdit->isHidden()) {
-                mTriggerPatternEdit[i]->lineEdit->show();
+            mTriggerPatternEdit[i]->lineEdit_pattern->clear();
+            if (mTriggerPatternEdit[i]->lineEdit_pattern->isHidden()) {
+                mTriggerPatternEdit[i]->lineEdit_pattern->show();
             }
-            mTriggerPatternEdit[i]->fgB->hide();
-            mTriggerPatternEdit[i]->bgB->hide();
-            mTriggerPatternEdit[i]->patternType->setCurrentIndex(0);
+            mTriggerPatternEdit[i]->pushButton_fgColor->hide();
+            mTriggerPatternEdit[i]->pushButton_bgColor->hide();
+            mTriggerPatternEdit[i]->comboBox_patternType->setCurrentIndex(0);
         }
+        // Scroll to the last used pattern:
+        mpScrollArea->ensureWidgetVisible(mTriggerPatternEdit.at(qBound(0, patternList.size(), 49)));
         QString command = pT->getCommand();
         mpTriggersMainArea->lineEdit_trigger_name->setText(pItem->text(0));
-        mpTriggersMainArea->trigger_command->setText(command);
+        mpTriggersMainArea->lineEdit_trigger_command->setText(command);
         mpTriggersMainArea->checkBox_multlinetrigger->setChecked(pT->isMultiline());
         mpTriggersMainArea->perlSlashGOption->setChecked(pT->mPerlSlashGOption);
         mpTriggersMainArea->filterTrigger->setChecked(pT->mFilterTrigger);
         mpTriggersMainArea->spinBox_linemargin->setValue(pT->getConditionLineDelta());
         mpTriggersMainArea->spinBox_stayOpen->setValue(pT->mStayOpen);
         mpTriggersMainArea->soundTrigger->setChecked(pT->mSoundTrigger);
         mpTriggersMainArea->lineEdit_soundFile->setText(pT->mSoundFile);
-        QPalette pal = palette();
-        QColor color = pal.color(QPalette::Button);
-        QString styleSheet = QString("QPushButton{background-color:") + color.name() + QString(";}");
-
-        QColor fgColor = pT->getFgColor();
-        QColor bgColor = pT->getBgColor();
-        QPalette FgColorPalette;
-        QPalette BgColorPalette;
-        FgColorPalette.setColor(QPalette::Button, fgColor);
-        BgColorPalette.setColor(QPalette::Button, bgColor);
-        QString FgColorStyleSheet = QString("QPushButton{background-color:") + fgColor.name() + QString(";}");
-        QString BgColorStyleSheet = QString("QPushButton{background-color:") + bgColor.name() + QString(";}");
-        mpTriggersMainArea->pushButtonFgColor->setStyleSheet(FgColorStyleSheet);
-        mpTriggersMainArea->pushButtonBgColor->setStyleSheet(BgColorStyleSheet);
-        mpTriggersMainArea->pushButtonFgColor->setPalette(FgColorPalette);
-        mpTriggersMainArea->pushButtonBgColor->setPalette(BgColorPalette);
```

Comment:
Now this is, apparently _naughty_ - see previous comment about mixing palettes and styles. ☹️